docs/src/integration_test_tool.md

```diff
@@ -1,3 +1,161 @@
+
+The crucial question is which `Project.toml` should be used. The dependency graph of a `Project.toml` can only be constructed in one direction. You can only use the dependencies of the packages. From which package a package is used can only be determined if you came from the package before. Let's assume we have the following package ecosystem:
+
+```mermaid
+graph TD
+    MyPkgMain -->|using| MyPkgA
+    MyPkgMain -->|using| MyPkgB -->|using| MyPkgD
+    MyPkgB -->|using| MyPkgE -->|using| MyPkgF
+    MyPkgMain -->|using| MyPkgC
+```
+
+If we take the `Project.toml`[\*] of `MyPkgMain`, we get the complete graph. If we take the `Project.toml` of `MyPkgB`, we get the following graph:
+
+```mermaid
+graph TD
+    MyPkgB -->|using| MyPkgD
+    MyPkgB -->|using| MyPkgE -->|using| MyPkgF
+```
+
+!!! note
+
+    The dependency tree that is created with the `Project.toml` of `MyPkgMain` does not contain `MyPkgMain` itself. This is a technical detail from Julia. It will contain the dependencies `MyPkgA` to `MyPkgF`. To solve the problem, an additional package is needed. The additional package only has the dependency `MyPkgMain`:
+
+    `Project.toml`
+    ```yaml
+    name = "MetaTestPkg"
+    uuid = "b849e8ad-e76f-4e2e-b89d-52f4e57509ba"
+    authors = ["Simeon Ehrig <s.ehrig@hzdr.de>"]
+    version = "0.1.0"
+
+    [deps]
+    MyPkgMain = "679daff0-1f79-468a-9a2e-69c3994cafb1"
+    ```
+
+    If we use the `Project.toml` of `MetaTestPkg`, we get the complete dependency graph with `MyPkgMain`.
+
+!!! note
+
+    To select a package and create the dependency graph, you need to activate the Julia environment of the package with `Pkg.activate()`. For example, `Pkg.activate("path/to/MetaTestPkg")`.
+
+## Project Dependency Tree: Special cases
+
+Let's assume we have the following package ecosystem structure:
+
+```mermaid
+graph TD
+    MyPkgA -->|using| MyPkgD
+    MyPkgB -->|using| MyPkgD
+    MyPkgB -->|using| MyPkgE
+    MyPkgC -->|using| MyPkgE
+    MyPkgC -->|using| MyPkgF
+    MyPkgD -->|using| MyPkgG
+    MyPkgD -->|using| MyPkgH
+    MyPkgE -->|using| MyPkgI
+```
+
+There is no package with which we can build the entire dependency graph. Therefore, an auxiliary package is required that is only created for generating the integration tests:
+
+```mermaid
+graph TD
+    MetaTestPkg -->|using| MyPkgA
+    MetaTestPkg -->|using| MyPkgB
+    MetaTestPkg -->|using| MyPkgC
+    MyPkgA -->|using| MyPkgD
+    MyPkgB -->|using| MyPkgD
+    MyPkgB -->|using| MyPkgE
+    MyPkgC -->|using| MyPkgE
+    MyPkgC -->|using| MyPkgF
+    MyPkgD -->|using| MyPkgG
+    MyPkgD -->|using| MyPkgH
+    MyPkgE -->|using| MyPkgI
+```
+
+If we use the `Project.toml` of `MetaTestPkg`, we can construct the entire graph.
+
+!!! note
+
+    The "MetaTestPkg" does not need to be registered in a registry. It can be saved locally in the project repository or created dynamically during the generation of the integration tests.
```

Comment:
```suggestion
    The "MetaTestPkg" does not need to be registered. It can be saved locally in the project repository or created dynamically during the generation of the integration tests.
```